File: M13-PROJECT/PycharmProjects/M09-PSP-UF1_Act1/classes/cesar.py
from dataclasses import dataclass


def alphabet():
    return "abcdefghijklmnopqrstuvwxyz"


@dataclass
class Cesar:
    key: int
    value: str

    def encryptCesar(self):
        return self.value.translate(str.maketrans(alphabet(), alphabet()[self.key:] + alphabet()[:self.key]))

    def decryptCesar(self):
        return self.value.translate(str.maketrans(alphabet()[self.key:] + alphabet()[:self.key], alphabet()))

File: M09-PSP/UF1/Xifratge/M09-PSP-UF1_Act1_java/src/main/java/daniel/Cabrera/classes/Cesar.java
package daniel.Cabrera.classes;

public class Cesar {
    private int key;
    private String value;
    private String alphabet = "abcdefghijklmnopqrstuvwxyz";

    public Cesar(int key, String value) {
        this.key = key;
        this.value = value;
    }

    public String encryptCesar() {
        StringBuilder encrypted = new StringBuilder();
        for (char c : value.toCharArray()) {
            if (Character.isLetter(c)) {
                char base = Character.isLowerCase(c) ? 'a' : 'A';
                encrypted.append((char) (base + (c - base + key) % 26));
            } else {
                encrypted.append(c);
            }
        }
        return encrypted.toString();
    }

    public String decryptCesar() {
        StringBuilder decrypted = new StringBuilder();
        for (char c : value.toCharArray()) {
            if (Character.isLetter(c)) {
                char base = Character.isLowerCase(c) ? 'a' : 'A';
                decrypted.append((char) (base + (c - base - key + 26) % 26));
            } else {
                decrypted.append(c);
            }
        }
        return decrypted.toString();
    }
}
